### ISSUE TITLE
query relation error

### ISSUE THREAD

**Issue by jaysneg** [amberframework/granite#209]
if add this method to Category model
```
has_many :subcategories

def subcategories
  Granite::AssociationCollection(self, Subcategory).new(self)
end
```

```
related to [https://github.com/amberframework/granite/issues/208](https://github.com/amberframework/granite/issues/208)

ERROR: Unknown column 'subcategories.categorie_id' in 'where clause' (Exception)
  from lib/mysql/src/mysql/connection.cr:95:5 in 'handle_err_packet'
  from lib/mysql/src/mysql/connection.cr:112:5 in 'raise_if_err_packet'
  from lib/mysql/src/mysql/statement.cr:20:7 in 'initialize'
  from lib/mysql/src/mysql/statement.cr:4:3 in 'new'
  from lib/mysql/src/mysql/connection.cr:151:5 in 'build_prepared_statement'
  from lib/db/src/db/connection.cr:7:15 in 'fetch_or_build_prepared_statement'
  from lib/db/src/db/session_methods.cr:58:9 in 'build'
  from lib/db/src/db/pool_prepared_statement.cr:37:16 in 'build_statement'
  from lib/db/src/db/pool_prepared_statement.cr:53:22 in 'initialize'
  from lib/db/src/db/pool_prepared_statement.cr:11:5 in 'new'
  from lib/db/src/db/database.cr:89:7 in 'build_prepared_statement'
  from lib/db/src/db/database.cr:7:15 in 'fetch_or_build_prepared_statement'
  from lib/db/src/db/session_methods.cr:23:9 in 'build'
  from lib/db/src/db/query_methods.cr:38:7 in 'query'
  from lib/granite/src/granite/querying.cr:53:7 in 'raw_all'
  from ~procProc(Array(Subcategory))@lib/granite/src/granite/querying.cr:74
  from lib/granite/src/granite/collection.cr:255:3 in 'collection'
  from src/controllers/home_controller.cr:182:5 in 'index'
  from ~procProc(HTTP::Server::Context, Nil)@config/routes.cr:37
  from lib/amber/src/amber/router/route.cr:255:3 in 'call'
  from lib/amber/src/amber/router/context.cr:77:5 in 'process_request'
  from lib/amber/src/amber/pipes/controller.cr:8:9 in 'call'
  from /usr/local/Cellar/crystal-lang/0.24.2_1/src/http/server/handler.cr:24:7 in 'call_next'
  from src/pipes/authenticate.cr:12:7 in 'call'
```

**Comment by faustinoaq** [comment]
Hi @jaysneg Thanks for report all these issue. 

Please use tiple backtricks to add code blocks, otherwise is hard to read your code snippets :sweat_smile:

Try this:

~~~
```
This is a code block
```
~~~

**Comment by jaysneg** [comment]
@faustinoaq ok

**Comment by robacarp** [comment]
This looks like a problem with pluralization, #129. I'm guessing your column name is `category_id` instead of `categorie_id`

**Comment by jaysneg** [comment]
yes

**Comment by conradwt** [comment]
I'm seeing a smilar issue where my join table, `friendships`, has a `person_id` and `friend_id`.  For example, it's generating the following invalid SQL:

```
SELECT "friends"."id", "friends"."person_id", "friends"."created_at","friends"."updated_at" FROM "friends" JOIN friendships ON friendships.friend_id = friends.id WHEREfriendships.peopl_id = $1: 25
```

Here's the SQL it should be generating:

```
SELECT "friends"."id", "friends"."person_id", "friends"."created_at","friends"."updated_at" FROM "friends" JOIN friendships ON friendships.friend_id = friends.id WHEREfriendships.person_id = $1: 25
```

**Comment by drujensen** [comment]
This will be addressed with #103 and #129
